### PR TITLE
made semantic vector a value class

### PR DESCRIPTION
**Pull request by ymatsuda**

### CODE REVIEW

**Comment by eishay** [comment]
cool!

File: shoebox/app/com/keepit/controllers/SearchLabsController.scala [hunk 1]
```diff
@@ -132,10 +132,10 @@ object SearchLabsController extends FortyTwoController {
     Ok(JsObject(Seq("data" -> JsArray(data))))
   }
 
-  private def similarity(vectors1: Array[Array[Byte]], vectors2: Array[Array[Byte]]) = {
+  private def similarity(vectors1: Array[SemanticVector], vectors2: Array[SemanticVector]) = {
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/SearchLabsController.scala:135]: Yeah, nice change. Unfortunately, as you probably know (and I learned just yesterday), putting value classes into arrays is one of the things that forces them to be instantiated.

Comment (ymatsuda) [inline shoebox/app/com/keepit/controllers/SearchLabsController.scala:135]: gee. I didn't know that!

Comment (eishay) [inline shoebox/app/com/keepit/controllers/SearchLabsController.scala:135]: didn't know that as well

http://docs.scala-lang.org/overviews/core/value-classes.html

> A value class is actually instantiated when:
> a value class is treated as another type.
> a value class is assigned to an array.
> doing runtime type tests, such as pattern matching.

would it have a significant performance impact in this case?

Comment (ymatsuda) [inline shoebox/app/com/keepit/controllers/SearchLabsController.scala:135]: this is an experimental code (friend map). so, it is ok.

**Pull request merged**